core/src/main/java/nu/yona/server/analysis/entities/DayActivity.java

```diff
@@ -144,22 +139,22 @@ private int getDurationInSpreadItem(ZonedDateTime startTime, ZonedDateTime endTi
 	{
 		if (spreadStartIndex == spreadEndIndex)
 		{
-			// partial span
-			return (int) startTime.until(endTime, ChronoUnit.MINUTES) + 1;
+			// partial span of spread item
+			return (int) startTime.until(endTime, ChronoUnit.MINUTES);
 		}
 		else if (spreadItemIndex == spreadStartIndex)
 		{
-			// start part
+			// start spread item
 			return 15 - (startTime.getMinute() % 15);
 		}
 		else if (spreadItemIndex == spreadEndIndex)
 		{
-			// end part
-			return 1 + (endTime.getMinute() % 15);
+			// end spread item
```

Comment:
`// spans begin of spread item till activity end point (which occurs inside spread item)`